services/port_manager/src/main/java/com/futurewei/alcor/portmanager/processor/DatabaseProcessor.java

```diff
@@ -22,19 +22,26 @@
 import java.util.*;
 
 public class DatabaseProcessor extends AbstractProcessor {
-    private NeighborInfo buildNeighborInfo(InternalPortEntity internalPortEntity) {
+    private List<NeighborInfo> buildNeighborInfos(InternalPortEntity internalPortEntity) {
+        List<NeighborInfo> neighborInfos = new ArrayList<>();
         String bindingHostIp = internalPortEntity.getBindingHostIP();
         if (bindingHostIp == null) {
-            return null;
+            return neighborInfos;
         }
 
-        NeighborInfo neighborInfo = new NeighborInfo(bindingHostIp,
-                internalPortEntity.getBindingHostId(),
-                internalPortEntity.getId(),
-                internalPortEntity.getMacAddress(),
-                internalPortEntity.getFixedIps().get(0).getIpAddress());
+        //TODO: Support a port with multiple neighborInfos
```

Comment:
Does this TODO get implemented already?

Reply:
Not yet

Reply:
Correct me if I am wrong. neighborInfos itself is an array and could contain more than one items assuming that a port has multiple fixed ips. 

Can you elaborate a bit what multiple neighbors you refer to? Which scenario may not be supported as this time.

Reply:
Oh, yes, neighborInfos itself is an array,  this comment is meaningless. Let me delete it.